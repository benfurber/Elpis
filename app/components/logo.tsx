import React from "react";
import { View, StyleSheet, Text } from "react-native";
import FlexImage from "react-native-flex-image";
import { colours, layout } from "styles";

const logoFile = "../assets/images/logo.png";

const Logo = () => {
  return (
    <View style={styles.container}>
      <View style={styles.textBox}>
        <FlexImage source={require(logoFile)} style={styles.image} />
        <Text style={styles.text}>Elpis</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: "flex-end",
    flex: 1,
    flexDirection: "row",
  },
  image: {
    borderRadius: 20,
    height: 40,
    overflow: "hidden",
    width: 40,
  },
  text: {
    alignSelf: "flex-end",
    color: colours.darkGrey,
    fontFamily: "Crete Round",
    fontSize: 32,
    marginLeft: layout.spacing,
    textAlign: "center",
  },
  textBox: {
    alignItems: "center",
    alignSelf: "flex-end",
    flex: 1,
    flexDirection: "row",
    justifyContent: "center",
    marginBottom: layout.spacing,
    width: "100%",
  },
});

export { Logo };
