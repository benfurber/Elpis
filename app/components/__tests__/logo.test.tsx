import "react-native";
import React from "react";
import { shallow } from "enzyme";

import { Logo } from "../logo";

describe("Logo", () => {
  it("renders correctly", () => {
    const component = shallow(<Logo />);

    expect(component).toMatchSnapshot();
  });
});
