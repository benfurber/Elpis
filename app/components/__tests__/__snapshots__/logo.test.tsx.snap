// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-end",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 10,
        "width": "100%",
      }
    }
  >
    <FlexImage
      source={
        Object {
          "testUri": "../../../app/assets/images/logo.png",
        }
      }
      style={
        Object {
          "borderRadius": 20,
          "height": 40,
          "overflow": "hidden",
          "width": 40,
        }
      }
    />
    <Text
      style={
        Object {
          "alignSelf": "flex-end",
          "color": "rgb(51, 51, 51)",
          "fontFamily": "creteround-regular",
          "fontSize": 32,
          "marginLeft": 10,
          "textAlign": "center",
        }
      }
    >
      Elpis
    </Text>
  </View>
</View>
`;
